communication/src/TlsConnectionManager.cpp

```diff
@@ -23,7 +23,62 @@ using namespace std::placeholders;
 
 namespace bft::communication::tls {
 
-void setSocketOptions(boost::asio::ip::tcp::socket& socket) { socket.set_option(boost::asio::ip::tcp::no_delay(true)); }
+void ConnectionManager::disableKeepAlive(boost::asio::ip::tcp::socket& socket,
+                                         const std::string& option,
+                                         int rc,
+                                         logging::Logger& logger) {
+  LOG_WARN(logger, "Set of " << option << " socket option failed; disabling socket keep-alive" << KVLOG(rc));
+  setKeepAliveOption(socket, false);
+}
+
+void ConnectionManager::setKeepAliveOption(boost::asio::ip::tcp::socket& socket, bool value) {
+  boost::asio::socket_base::keep_alive keepAlive(value);
+  socket.set_option(keepAlive);
+}
+
+void ConnectionManager::enableKeepAlive(boost::asio::ip::tcp::socket& socket,
+                                        const TcpKeepAliveConfig& keepAliveConfig,
+                                        logging::Logger& logger) {
+  setKeepAliveOption(socket, true);
+
+  int idleTime = keepAliveConfig.socketKeepAliveIdleTime;
+  uint32_t optionLen = sizeof(idleTime);
+  auto rc = setsockopt(socket.native_handle(), IPPROTO_TCP, TCP_KEEPIDLE, &idleTime, optionLen);
+  if (rc != 0) {
+    return disableKeepAlive(socket, "TCP_KEEPIDLE", rc, logger);
+  }
+  getsockopt(socket.native_handle(), IPPROTO_TCP, TCP_KEEPIDLE, &idleTime, &optionLen);
```

Comment:
why do you need the getsocket call?

Reply:
just for the log in the end?

Reply:
I want to verify that the parameters have been really set as required.